### PR TITLE
Host connection settings UI

### PR DESCRIPTION
**Pull request by nelsonkopliku**
image

### CODE REVIEW

File: assets/js/components/ClusterDetails/ConnectionSettings.jsx [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+              payload: {
+                cluster: cluster?.id,
+                settings: ls,
+              },
+            });
+            // dispatch({
+            //     type: 'CHECKS_SELECTED',
+            //     payload: { checks: selectedChecks, clusterID: clusterID },
+            // });
+            // navigate(`/clusters/${clusterID}/checks/results`);
```
Comment (dottorblaster) [inline assets/js/components/ClusterDetails/ConnectionSettings.jsx:167]: :eyes: could we strip this?

File: assets/js/components/ClusterDetails/ConnectionSettings.jsx [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+          }}
+        >
+          {saving ? (
+            <span className="px-20">
+              <EOS_LOADING_ANIMATED color="green" size={25} />
+            </span>
```
Comment (dottorblaster) [inline assets/js/components/ClusterDetails/ConnectionSettings.jsx:173]: We already have a Spinner component if I'm not wrong, keep in mind

File: assets/js/components/ClusterDetails/ConnectionSettings.jsx [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+            <button
+              className="absolute top-0 bottom-0 right-0 pr-2"
+              onClick={() => setLocalSavingError(null)}
+            >
+              <EOS_CANCEL size={14} className="fill-red-600" />
+            </button>
+          </div>
+        )}
+      </div>
+    </div>
```
Comment (dottorblaster) [inline assets/js/components/ClusterDetails/ConnectionSettings.jsx:193]: For the future: this could be a component inside the design system

**Review by dottorblaster — CHANGES_REQUESTED** [review]
Some changes to apply, then we are good to merge

File: assets/js/components/ClusterDetails/ConnectionSettings.jsx [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+                className={classNames(
+                  item.isDefaultUser ? 'bg-jungle-green-500' : 'bg-gray-200',
+                  'relative inline-flex flex-shrink-0 h-6 w-11 border-2 border-transparent rounded-full cursor-pointer focus:outline-none transition-colors ease-in-out duration-200'
+                )}
+                onChange={() => toggleUseDefaultConnectionUser(item.host_id)}
+              >
+                <span
+                  aria-hidden="true"
+                  className={classNames(
+                    item.isDefaultUser ? 'translate-x-5' : 'translate-x-0',
```
[the remark this answers is not in the document]
Comment (nelsonkopliku) [reply assets/js/components/ClusterDetails/ConnectionSettings.jsx:129]: Thanks!

File: assets/js/components/ClusterDetails/ConnectionSettings.jsx (hunk 1 above)
Comment (nelsonkopliku) [reply assets/js/components/ClusterDetails/ConnectionSettings.jsx:167]: Next, need for the next iteration! :sweat_smile:

File: assets/js/components/ClusterDetails/ConnectionSettings.jsx (hunk 2 above)
Comment (nelsonkopliku) [reply assets/js/components/ClusterDetails/ConnectionSettings.jsx:173]: Couldn't make it work properly. Gonna keep it like this for now :sweat_smile:

**Review by dottorblaster — APPROVED** [review]

**Pull request merged**